src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -821,7 +821,7 @@ void PersistNode(TrieNode n)
             });
         }
 
-        public byte[]? this[byte[] key]
+        public byte[]? this[ReadOnlySpan<byte> key]
```

Comment:
If span is used here, have you considered using the `SpanConcurrentDictionary` for `AllNodes` to not alloc the keccak at all?

Reply:
Not sure what you mean, AllNodes returns the cache directly. Maybe it was recently changed.

Reply:
Sorry. I placed the comment under wrong line. Adding a separate one and resolving this.